internal/registry/registry.go

```diff
@@ -1106,6 +1110,26 @@ func (r *Registry) CheckSequence(peerID peer.ID, seq uint64) error {
 	return r.sequences.check(peerID, seq)
 }
 
+// periodicProviderReload reloads the registry's providers from the datastore.
+func (r *Registry) periodicProviderReload(interval time.Duration) {
+	ticker := time.NewTicker(interval)
+	for {
+		select {
+		case <-r.closing:
+			return
+		case <-ticker.C:
+			providers, err := loadPersistedProviders(context.Background(), r.dstore, r.filterIPs)
+			if err != nil {
+				log.Errorw("cannot reload provider data from datastore", "err", err)
```

Comment:
Leave the stale data there?

```suggestion
				log.Errorw("cannot reload provider data from datastore", "err", err)
				return
```